### ISSUE TITLE
Is there a way to display multiple box annotation colours for different object detections based on multiple tracker ids?

### ISSUE THREAD

**Issue by ElNoSabe322** [roboflow/supervision#172]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Question

First of all, thank you so much for Supervision. It is amazing. <br>
So, in my object detection tracking, sometimes I track multiple objects and sometimes I track only one object. However, regardless of the case, I want to be able to display a unique bounding box colour based on the corresponding tracking ID. 
As far as I know, there exists a method called ```by_idx()``` which returns a unique colour for one ID only. Is there a way to include multiple IDs for this?<br>
```python
# Defining the bounding box annotator
box_annotator = sv.BoxAnnotator(
     color=sv.ColorPalette.default().by_idx(
        idx=tracker_detections.tracker_id
     ),
     thickness=3,
     text_color=sv.Color.white(),
     text_scale=0.8,
     text_thickness=1,
     text_padding=10,
)
``` 
```tracker_detections``` is a supervision Detections object.
Yes, I am getting an error in this code since ```by_idx()``` only supports a single integer and not an array of integers. Is there any workaround for this?

**Comment by hardikdava** [comment]
@ElNoSabe322 Thank you for reaching out. Currently, `sv.BoxAnnotator` supports either single color to all instances or color based on `class id` i.e. `by_idx(class_id)`. Thank for your suggestion to consider `color by track id`. We will consider it. It is highly recommended to visualize object trajectories with different color. We are currently working on improving different `Annotators`. You can check the progress #170. Stay tune for more updates!

**Comment by ElNoSabe322** [comment]
@hardikdava Thank you for your reply! Yeah, I'm really looking forward to the new re-designed annotators, especially with the trail track annotation and the box mask annotation. Can't wait for the new Supervision version. 😁👏

**Comment by hardikdava** [comment]
@ElNoSabe322 You might be interested in  #49 . FYI: this feature will be available very soon.

**Comment by ElNoSabe322** [comment]
Yeah, I cannot wait to see this feature in action. This will be a lifesaver!! 😌🙏

**Comment by ElNoSabe322** [comment]
I found a temporary workaround for this issue until the new update comes. 
Basically, I made a separate supervision detection class object for each tracked object and in this way I can pass one ```tracker_id``` at a time. Therefore, for ```n``` tracked objects, there will be ```n``` detection class objects and ```n``` box annotators and ```n``` annotate function calls. 
This is not time and memory efficient especially if ```n``` is large so yeah... looking forward to the new update. 🙏

@hardikdava should I close this issue or should I leave it open until the update includes this?

**Comment by hardikdava** [comment]
@ElNoSabe322 Thank you for your comments. You can leave this issue open. We will let you know when this feture is implemented.

**Comment by SkalskiP** [comment]
Hi @ElNoSabe322 and @hardikdava, I'm doing supervision housekeeping. I'm converting this question into a discussion and moving it to the QA section.
